集成测试失败: ESM 构建不支持 form-data 的动态 require

## 测试输出错误信息

```
[ERROR] Server failed to start within 30 seconds

Error: Dynamic require of "util" is not supported
    at file:///Users/hs3180/clawd/disclaude/dist/chunk-VZ5GRZJO.js:12:9
    at node_modules/combined-stream/lib/combined_stream.js
    at node_modules/form-data/lib/form_data.js
    at ModuleJob.run (node:internal/modules/esm/module_job:371:25)
    at async importRunners (file:///Users/hs3180/clawd/disclaude/dist/cli-entry.js:138:19)
    at async main (file:///Users/hs3180/clawd/disclaude/dist/cli-entry.js:225:47)
```

## 根本原因分析（来自日志）

**问题**: tsup 构建 ESM 产物时，未能正确处理 `form-data` 及其依赖 `combined-stream` 的 CommonJS 动态 require。

**技术细节**:
1. `form-data` 是一个 CommonJS 模块，内部使用 `require('util')`
2. tsup 打包时将其内联到 `chunk-AJURUO6N.js`
3. 运行时 ESM loader 不支持这种动态 require 模式
4. 导致服务启动时崩溃

**影响范围**: 
- 集成测试无法启动测试服务器
- 任何使用 `dist/cli-entry.js` 的生产部署都会失败

## 相关日志

```json
{
  "type": "_AppError",
  "message": "Dynamic require of \"util\" is not supported",
  "stack": "Error: Dynamic require of \"util\" is not supported
    at file:///Users/hs3180/clawd/disclaude/dist/chunk-VZ5GRZJO.js:12:9
    at node_modules/combined-stream/lib/combined_stream.js
    at node_modules/form-data/lib/form_data.js",
  "category": "UNKNOWN",
  "severity": "error",
  "retryable": false
}
```

## 建议修复方向

1. 在 `tsup.config.ts` 中将 `form-data` 添加到 `external` 列表
2. 或使用 `--format cjs` 替代 ESM
3. 或升级 `form-data` 到支持 ESM 的版本

## 触发信息

- 触发提交: 76681a0 (Merge pull request #527)
- 检测时间: 2026-03-03 16:02
- 测试类型: 集成测试 (npm run test:integration)

--- a/tsup.config.ts
+++ b/tsup.config.ts
@@ -13,6 +13,13 @@ const EXTERNAL_DEPS = [
   'pino-roll',
   'sonic-boom',
   'pino-file',
+  // CJS modules that use dynamic require - must be external for ESM build
+  // These are dependencies of @larksuiteoapi/node-sdk -> axios
+  'axios',
+  'form-data',
+  'follow-redirects',
+  'combined-stream',
+  'proxy-from-env',
 ];
 
 export default defineConfig([

